node/src/neighborhood/mod.rs

```diff
@@ -1434,6 +1693,230 @@ impl Neighborhood {
+        &self,
+        client_id: u64,
+        context_id: u64,
+        missing_countries: Vec<String>,
+        show_countries_flag: bool,
+    ) -> NodeToUiMessage {
+        let fallback_routing = match &self.user_exit_preferences.fallback_preference {
+            FallbackPreference::Nothing => true,
+            FallbackPreference::ExitCountryWithFallback => true,
+            FallbackPreference::ExitCountryNoFallback => false,
+        };
+        let exit_locations = self
+            .user_exit_preferences
+            .locations_opt
+            .clone()
+            .unwrap_or_default();
+        let show_countries = match show_countries_flag {
+            true => Some(self.user_exit_preferences.db_countries.clone()),
+            false => None,
+        };
+
+        if !self.error_message_indicates(&mut missing_countries.clone()) {
+            NodeToUiMessage {
+                target: MessageTarget::ClientId(client_id),
+                body: UiSetExitLocationResponse {
+                    fallback_routing,
+                    exit_locations,
+                    exit_countries: show_countries,
+                    missing_countries,
+                }
+                .tmb(context_id),
+            }
+        } else {
+            let missing_message: String = missing_countries.join(", ");
+            NodeToUiMessage {
+                target: MessageTarget::ClientId(client_id),
+                body: MessageBody {
+                    opcode: "exitLocation".to_string(),
+                    path: Conversation(context_id),
+                    payload: Err((EXIT_COUNTRY_MISSING_COUNTRIES_ERROR, missing_message)),
+                },
+            }
+        }
+    }
+
+    fn set_exit_locations_opt(&mut self, exit_locations_by_priority: &[ExitLocation]) {
+        self.user_exit_preferences.locations_opt =
+            match self.user_exit_preferences.exit_countries.is_empty() {
+                false => Some(exit_locations_by_priority.to_owned()),
+                true => match self.user_exit_preferences.fallback_preference {
+                    FallbackPreference::ExitCountryNoFallback => None,
+                    _ => Some(exit_locations_by_priority.to_owned()),
+                },
+            };
+    }
+
+    fn set_country_undesirability_and_exit_countries(
+        &mut self,
+        exit_locations_by_priority: &Vec<ExitLocation>,
+    ) {
+        let nodes = self.neighborhood_database.nodes_mut();
+        match !&exit_locations_by_priority.is_empty() {
+            true => {
+                for node_record in nodes {
+                    self.user_exit_preferences
+                        .assign_nodes_country_undesirability(node_record)
+                }
+            }
+            false => {
+                self.user_exit_preferences.exit_countries = vec![];
+                for node_record in nodes {
+                    node_record.metadata.country_undesirability = 0u32;
```

Comment:
Prosim vytvor konstantu i pro tuhle nulovou hodnotu. At to jasne vyjadruje, ze to je promyslena, volena hodnota